March news (#741)
ropensci/roweb3: content/blog/2024-03-22-news-march-2024/index.Rmd

---
title: 'rOpenSci News Digest, March 2024'
author: 'The rOpenSci Team'

> date: '2024-03-22'

Reviewer: Just a reminder to update the dates here and in the folder 😁

You can read this post [on our blog](`r url`).
Now let's dive into the activity at and around rOpenSci!

## rOpenSci HQ

### rOpenSci Dev Guide 0.9.0: Multilingual Now! And Better

We're delighted to announce we've released a new version of our guide,
["rOpenSci Packages: Development, Maintenance, and Peer Review"](https://devguide.ropensci.org/)!

A highlight is that our guide is now bilingual (English and [Spanish](https://devguide.ropensci.org/es/index.es.html)), thanks to work by [Yanina Bellini Saibene](/author/yanina-bellini-saibene/), [Elio Campitelli](/author/elio-campitelli/) and [Pao Corrales](/author/pao-corrales/), and thanks to [support of the Chan Zuckerberg Initiative, NumFOCUS, and the R Consortium](/blog/2023/01/12/multilingual-publishing-en/).
[Read the guide in Spanish](https://devguide.ropensci.org/es/index.es.html).

Our guide is now also getting translated to [Portuguese](/blog/2023/10/20/news-october-2023/#help-us-translate-our-dev-guide-to-portuguese) thanks to volunteers.
We are very grateful for their work!

Read more in the [blog post about the release](/blog/2024/03/11/devguide-0.9.0/).
Thanks to all contributors who made this release possible.

> ### Interview of our community manager Yanina Bellini Saibene

Reviewer: ```suggestion
### Interview with Code for Thought podcast
```

> Yanina talked with Peter Schmidt of the [Code for thought podcast](https://codeforthought.buzzsprout.com/1326658/14660478-en-crossing-the-language-barrier-yanina-saibene), about the importance to make computing materials accessible to non-English speaking learners. 

Reviewer: ```suggestion
Our community manager, Yanina Bellini Saibene, talked with Peter Schmidt of the [Code for Thought podcast](https://codeforthought.buzzsprout.com/1326658/14660478-en-crossing-the-language-barrier-yanina-saibene), about the importance of making computing materials accessible to non-English speaking learners. 
```

if (length(tech_notes) > 0) {
  cat("\n\n")
  cat("### Tech Notes\n\n")
  cat(
    paste0(
      purrr::map_chr(tech_notes, format_post),
      collapse = "\n\n"
    )
  )
  cat("\n\n")
}
```

## Calls for contributions

### Calls for maintainers

If you're interested in maintaining any of the R packages below, you might enjoy reading our blog post [What Does It Mean to Maintain a Package?](/blog/2023/02/07/what-does-it-mean-to-maintain-a-package/).

[internetarchive](https://docs.ropensci.org/internetarchive/), an API Client for the Internet Archive. [Issue for volunteering](https://github.com/ropensci/internetarchive/issues/17).

[historydata](https://docs.ropensci.org/historydata/), datasets for historians. [Issue for volunteering](https://github.com/ropensci/historydata/issues/23).

[textreuse](https://docs.ropensci.org/textreuse/), detect text reuse and document similarity. [Issue for volunteering](https://github.com/ropensci/textreuse/issues/97).

[tokenizers](https://docs.ropensci.org/tokenizers/), fast, consistent tokenization of natural language text. [Issue for volunteering](https://github.com/ropensci/tokenizers/issues/86).

> [USAboundaries](https://docs.ropensci.org/USAboundaries/) (and USAboundariesdata), historical and contemporary boundaries of the United States of America . [Issue for volunteering](https://github.com/ropensci/USAboundaries/issues/50).

Reviewer: Bulleted list?

### Calls for contributions

[Help make qualtRics better! Code contributions wanted](/blog/2024/03/01/qualtrics-call-help/)

[Help make assertr better! Come close issues](/blog/2024/02/27/assertr-call-help/)

[Help users and developers of targets by answering questions!](/blog/2024/02/29/targets-call-help/)

> [Help make waywiser better! User requests wanted](/blog/2024/02/19/waywiser-call-help/)

Reviewer: Bulleted list?

Also refer to our [help wanted page](/help-wanted/) -- before opening a PR, we recommend asking in the issue whether help is still needed.

> ## Package development corner

Reviewer: Okay, I'm officially creating a News Tips note in my Workflowy to keep track of all the awesome tips I get from this section, thank you!

Author: :tada:

> seeing this pop up in a few Slacks so figured I'd share
> https://github.com/tidyverse/glue/issues/320

Thanks, Mike!

### Useful hack: a CRAN-specific .Rbuildignore

The [`.Rbuildignore` file](https://blog.r-hub.io/2020/05/20/rbuildignore/) lists the files to not be included when building your package, such as your pkgdown configuration file.
Trevor L. Davis posted a neat idea on [Mastodon](https://mastodon.social/deck/@trevorld@fosstodon.org/112012986709303927): using a CRAN-specific `.Rbuildignore`, so that CRAN submissions omit some tests and vignettes to keep the package under the size limit.

Regarding tests themselves, remember you can [skip some or all](https://books.ropensci.org/http-testing/cran-preparedness.html#skipping-a-few-tests-on-cran) on CRAN (but make sure you're running them on [continuous integration](https://devguide.ropensci.org/pkg_ci.html)!).

### Key advantages of using the keyring package

If your package needs the user to provide secrets, like API tokens, to work, you might be interested in wrapping or recommending the keyring package (maintained by Gábor Csárdi), that accesses the system credential store from R.

> See this recent [R-hub blog post](/2024/02/28/key-advantages-of-using-keyring/).

Reviewer: ```suggestion
See this recent [R-hub blog post](https://blog.r-hub.io/2024/02/28/key-advantages-of-using-keyring/).
```
